### ISSUE TITLE
Test is not identified if there is no named receiver.

### ISSUE THREAD

**Issue by SimonRichardson** [babakks/vscode-go-test-suite#12]
If the suite has no named receiver then it doesn't parse the test correctly.

```
import (
    gc "gopkg.in/check.v1"
)

type suite struct{}

var _ = gc.Suite(&suite{})

func (*suite) TestMe(c *gc.C) {
}
```

**Comment by SimonRichardson** [comment]
One solution would be to change the following line:

https://github.com/babakks/vscode-go-test-suite/blob/a99b728c6bdd0c087d524385ba7b4cda8b2e752e/src/goParser.ts#L39C49-L39C50

To: `/^func \(.*?\*?(.*?)\) (Test.*?)\(.*? \*?(?:(.*?)\.)?(.*?)\) \{/`

Though, that means that this also matches `func (a b *suite) ...` which is not ideal, although that is also invalid go code, so I'm unsure if it really matters?

**Comment by babakks** [comment]
@SimonRichardson Thanks for reporting this. Yeah that change could match with other cases. How about the one below?

```
/^func \((.*? )?\*?(.*?)\) (Test.*?)\(.*? \*?(?:(.*?)\.)?(.*?)\) \{/
```

**Comment by SimonRichardson** [comment]
@babakks github gobbled up the regex, do want to put it in a code block so that doesn't happen?

**Comment by babakks** [comment]
Just edited.

**Comment by babakks** [comment]
I've tested it and it seems working. I'll now publish a new version of the extension.

**Comment by babakks** [comment]
@SimonRichardson I just published the new version, v0.0.11. Could you please upgrade and check if it works as expected?

**Comment by SimonRichardson** [comment]
It did - amazing!
